main.py

```diff
@@ -1,16 +1,153 @@
+import asyncio
+import base64
+import datetime
+import glob
+import json
 import os
+import ssl
+from asyncio import Lock
+
+import certifi
 import decky_plugin
+from aiohttp import ClientSession
 from settings import SettingsManager
 
+
 class Plugin:
+    yt_process: asyncio.subprocess.Process | None = None
+    # We need this lock to make sure the process output isn't read by two concurrent readers at once.
+    yt_process_lock = Lock()
+    music_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/music"
+    cache_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/cache"
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+
     async def _main(self):
-         self.settings = SettingsManager(name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR)
+        self.settings = SettingsManager(
+            name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR
+        )
 
     async def _unload(self):
-        pass
+        if self.yt_process is not None:
+            self.yt_process.terminate()
 
     async def set_setting(self, key, value):
         self.settings.setSetting(key, value)
 
     async def get_setting(self, key, default):
-        return self.settings.getSetting(key, default)
\ No newline at end of file
+        return self.settings.getSetting(key, default)
+
+    async def search_yt(self, term: str):
+        if self.yt_process is not None:
+            self.yt_process.terminate()
+            # Wait for process to terminate.
+            async with self.yt_process_lock:
+                await self.yt_process.communicate()
+        self.yt_process = await asyncio.create_subprocess_exec(
+            f"{decky_plugin.DECKY_PLUGIN_DIR}/yt-dlp",
```

Comment:
Binaries from remote_binary will end up in `f"{decky_plugin.DECKY_PLUGIN_DIR}/bin/yt-dlp"`